frontend/src/services/cartService.ts

```diff
@@ -0,0 +1,62 @@
+import { Storage } from '../utils/localStorage';
+
+export namespace CartService {
+  let cart = Storage.getItem(Storage.keys.cart) || [];
+  let listeners = [];
+
+  export const syncWithLS = () => {
+    listeners.forEach((listener) => listener(cart));
+    Storage.setItem(Storage.keys.cart, cart);
+    return Promise.resolve(cart);
+  };
+  export const addItem = (product) => {
+    const cartProducts = cart.some((item) => item.id === product.id);
```

Comment:
cart.some will return boolean, all boolean variables should starts from: has, is, can. ex: hasCartProduct

Reply:
use cart.find to find exact element, and after it you can modify this product